### PR TITLE
server-1557 | Enable GCP nomad module for workload identity

### PR DESCRIPTION
**Pull request by atulsingh0**
:gear: **Issue**

Enable [GCP Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) for Nomad Service account 

:white_check_mark: **Fix**

* Add `enable_workload_identity` input variable
* If `enable_workload_identity` true, it will add `roles/iam.workloadIdentityUser` in nomad service account considering GCP cluster is already enabled for `workload identity` 
* If `enable_workload_identity` true, Module will not generate the static keys for the service account 

:question: **Tests**

- [ ] Passed _reality check_

### CODE REVIEW

File: nomad-gcp/examples/basic/terraform.tfvars_template [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+project           = "project123"
+region            = "asia-south1"
+zone              = "asia-south1-b"
+network           = "default"
+subnetwork        = "default"
+min_replicas      = 1
+nomad_auto_scaler = true
+server_endpoint   = "nomad.exmaple.com:4647"
+enable_workload_identity = true
+machine_type      = "n2-standard-8"
```
Comment (ryanwohara) [inline nomad-gcp/examples/basic/terraform.tfvars_template:11]: Can we format this? HCL should have all the equal signs aligned when the lines are adjacent.

File: nomad-gcp/examples/basic/main.tf [hunk 2]
```diff
@@ -54,9 +94,11 @@ module "nomad" {
...
   # Autoscaling for Managed Instance Group
   autoscaling_mode  = "ON"
-  nomad_auto_scaler = false # If true, will generate a service account to be used by nomad-autoscaler. The is output in the file nomad-as-key.json
-  max_replicas      = 4     # Max and Min replica values should match the values intended to be used by nomad autoscaler in CircleCI Server
-  min_replicas      = 1
+  nomad_auto_scaler = var.nomad_auto_scaler # If true, will generate a service account to be used by nomad-autoscaler. The is output in the file nomad-as-key.json
+  max_replicas      = var.max_replicas      # Max and Min replica values should match the values intended to be used by nomad autoscaler in CircleCI Server
+  min_replicas      = var.min_replicas
+  enable_workload_identity = var.enable_workload_identity     # If using GCP work identities rather than static keys in CircleCI Server
+
```
Comment (ryanwohara) [inline nomad-gcp/examples/basic/main.tf:101]: Can we run a `terraform fmt` on this please?

Comment (atulsingh0) [reply nomad-gcp/examples/basic/main.tf:101]: sure

File: nomad-gcp/examples/basic/terraform.tfvars_template (hunk 1 above)
Comment (atulsingh0) [reply nomad-gcp/examples/basic/terraform.tfvars_template:11]: sure

**Review by soulchips — APPROVED** [review]

**Review by ryanwohara — APPROVED** [review]

**Pull request merged**